### PR TITLE
WIP - Remove Redux

### PR DESCRIPTION
**Pull request by samuelbeard**
Do not merge yet.

I like making PRs before they are complete so all the progress is visible and others can comment on it as it is developed.

Stuff to do:
- [x] Add userData request to `App.js`
- [x] Add UserContext to `App.js`
- [x] Add updateLogged and isLogged to `App.js` and to UserContext
- [x] Add token request to `Login.js`

### CODE REVIEW

**Comment by samuelbeard** [comment]
@Maximilian118 Can you list out anything else that redux CURRENTLY does that will need to be redone outside of redux?

**Comment by samuelbeard** [comment]
It's ready to review. There are a few things missing but I feel like I have stolen the whole project if I carry on here. We can just add issues for the rest of the stuff. 

Sign up, login, logout all work fine (With the new backend too)

**Comment by samuelbeard** [comment]
Tests will fail but ah well for this. I'll get on them once pulled.

File: src/components/Error.js [hunk 1]
```diff
@@ -1,31 +1,27 @@
...
 
 Error.defaultProps = {
-    sidebar: false,
     message: "An error has occurred.",
 }
 
 export default Error
+
+// TO DO
+// This component doesn't unmount when it is closed. What other way can we do this?
```
Comment (Maximilian118) [inline src/components/Error.js]: You need a cleanup function. Although the JSX is unmounted from the DOM anything state still lives on. I assume that's what's happening here. Please fix this.

Although you're not using useEffect here, the "effects with cleanup" section of this page I think explains what's going on here:
https://reactjs.org/docs/hooks-effect.html

File: src/index.js [hunk 2]
```diff
@@ -1,43 +1,18 @@
...
-    </Provider>,
+    <BrowserRouter>
+        <App />
+    </BrowserRouter>,
     document.getElementById('root')
 )
 
+// If you want your app to work offline and load faster, you can change
+// unregister() to register() below. Note this comes with some pitfalls.
+// Learn more about service workers: https://bit.ly/CRA-PWA
```
Comment (Maximilian118) [inline src/index.js]: lol... did you completely remake CRA?

File: src/pages/Home/Home.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    }
+
+    const handleSubmit = e => {
+        e.preventDefault()
+        console.log('SUBMITTING HOME FORM')
+        axios
+        .post(process.env.REACT_APP_BASE_API_ROUTE, {
+            query: `
+            mutation {
+                createUser(userInput: {email: "${form.email}", username: "${form.username}", name: "${form.name}", password: "${form.password}", confirm: "${form.confirm}"}) {
```
Comment (Maximilian118) [inline src/pages/Home/Home.js]: Variables needed.

File: src/pages/Home/Home.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                            <Card
+                                style={{
+                                    background: "white",
+                                    border: "none",
+                                }}
+                            >
+                                <CardBody>
+                                    <Form onSubmit={handleSubmit}>
+                                        <FormGroup>
+                                            <Label for="email">Email</Label>
```
Comment (Maximilian118) [inline src/pages/Home/Home.js]: I knew you were going to bin dynamic forms... *sigh*

File: src/pages/Home/request.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import axios from "axios"
+
+const request = form => {
+    axios
+        .post(process.env.REACT_APP_BASE_API_ROUTE, {
+            query: `
+            mutation {
```
Comment (Maximilian118) [inline src/pages/Home/request.js]: Variables needed.

File: src/pages/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+        })
+    }
+
+    const handleRequest = e => {
+        e.preventDefault()
+        axios
+            .post(process.env.REACT_APP_BASE_API_ROUTE, {
+                query: `
+                    query {
+                        login(email: "${form.email}", password: "${form.password}") {
```
Comment (Maximilian118) [inline src/pages/Login/Login.js]: Variables needed.

File: src/pages/Login/Login.js [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+                localStorage.setItem("token", obj.token)
+                updateLogged()
+                setForm({ email: "", password: "" })
+            })
+    }
+
+    // if (isLogged) {
+    //     updateLogged()
+    //     return <Redirect to="/" />
+    // }
```
Comment (Maximilian118) [inline src/pages/Login/Login.js]: Should we be pushing commented out code?

File: src/components/Error.js (hunk 1 above)
Comment (samuelbeard) [reply src/components/Error.js]: I'm talking about triggering the unmount, not handling it once it happens.

File: src/index.js (hunk 2 above)
Comment (samuelbeard) [reply src/index.js]: No. Not sure how this got back in lol. Maybe copied and pasted from history.

**Comment by samuelbeard** [comment]
Tests will probably fail. We now have access to GitHub actions so I want to run tests on that now instead of Travis. That will be a new PR.

**Pull request merged**